src/common/melody_generator.py

```diff
@@ -0,0 +1,134 @@
+import tensorflow.keras as keras
+import json
+import numpy as np
+import music21 as m21
+
+
+class MelodyGenerator:
+
+    # TODO: Proper dependency injection (is that even possible in Python?)
+    def __init__(self, model_path: str, music_helper, file_helper, mapping_path: str, sequence_length: int):
+        self.model = keras.models.load_model(model_path)
+        self._music_helper = music_helper
+        self._file_helper = file_helper
+        self._start_symbols = ["/"] * sequence_length # This acts as our song delimtter 
+
+        # TODO: move to file_helper
+        with open(mapping_path, "r") as fp:
+            self._mappings = json.load(fp)
+
+    
+    def generate_melody(self, seed, num_steps, max_seq_len, temperature):
+        """ 
+        * seed - what we want to pass to the network and the network countinues that seed
+        * num_steps - How far we want the network to predict
+        * max_seq_len - How many steps of the seeds do we want to consider for the network (the seed will grow large), 
+        so we pass the seed as a sliding window
+        """
```

Comment:
Consider using the format:
```
Parameters
------------
stuff - stuff

Returns
--------
stuff - stuff
```
to standardize comments about what the inputs and outputs are